connectivity/ble_weight_scale_nau7802/src/ble_weight_scale_nau7802.c

```diff
@@ -0,0 +1,146 @@
+ *
+ * @file ble_weight_scale_nau7802.c
+ *
+ * @brief BLE weight scale project source code.
+ *
+ * Copyright (C) 2012-2019 Dialog Semiconductor.
+ * This computer program includes Confidential, Proprietary Information
+ * of Dialog Semiconductor. All Rights Reserved.
+ *
+ ****************************************************************************************
+ */
+
+/**
+ ****************************************************************************************
+ * @addtogroup APP
+ * @{
+ ****************************************************************************************
+ */
+#include "rwip_config.h"             // SW configuration
+
+
+/*
+ * INCLUDE FILES
+ ****************************************************************************************
+ */
+
+#include "app_api.h"
+#include "ble_weight_scale_nau7802.h"
+#include "app_wsss.h"
+#include "nau7802.h"
+#include "custs1_task.h"
+#include "user_custs1_def.h"
+#include "arch_console.h"
+
+/*
+ * VARIABLE DEFINITIONS
+ ****************************************************************************************
+ */
+uint8_t conidx																				__SECTION_ZERO("retention_mem_area0"); //@RETENTION MEMORY
+timer_hnd timer_id																		__SECTION_ZERO("retention_mem_area0"); //@RETENTION MEMORY
+struct wss_wt_meas* meas;
+
+/*
+ * FUNCTION DEFINITIONS
+ ****************************************************************************************
+*/
+
+void user_on_connection(uint8_t connection_idx, struct gapc_connection_req_ind const *param)
+{
+		meas = (struct wss_wt_meas* ) ke_malloc(sizeof(struct wss_wt_meas),KE_MEM_ATT_DB);
+    memset(meas, 0, sizeof(struct wss_wt_meas));
+	
+		#ifdef CFG_PRINTF
+		arch_puts("Connection\n\r");
+		#endif
+		conidx = connection_idx;
+		app_wsss_enable(connection_idx);
+    default_app_on_connection(connection_idx, param);
+	
+		nau7802_init();
+		nau7802_enable_adc();
+	
+		timer_id = app_easy_timer(WEIGHT_SCALE_UPDATE_INTERVAL,update_weight);
+		
+}
+
+void user_on_disconnect( struct gapc_disconnect_ind const *param )
+{
+		app_easy_timer_cancel(timer_id);
+    default_app_on_disconnect(param);
+}
+
+void user_app_on_init( void )
+{
+		#ifdef CFG_PRINTF
+		arch_puts("User app init\n\r");
+		#endif
+		app_wsss_init();
+		app_wsss_create_db();
+}
+
+void update_weight( void )
+{
+		uint32_t load;
+		(void)nau7802_get_ug_load(&load);
+		meas->weight = ((float)load)/10000000; // convert 10 mg to kg
+		app_wsss_send_measurement(conidx, meas);
+		app_easy_timer(WEIGHT_SCALE_UPDATE_INTERVAL,update_weight);
+	
+		#ifdef CFG_PRINTF
+		arch_printf("Load: %d\n\r", (int)(meas->weight));
+		#endif
+}
+
+uint32_t grams_byte_array_to_10milligrams(const uint8_t data[4])
+{	
+		uint32_t dat = (uint32_t)((((uint32_t)(data[3])) << 24) |
+														  (((uint32_t)(data[2])) << 16)  |
+														  (((uint32_t)(data[1])) << 8)  |
+															 ((uint32_t)(data[0])));
+		return dat*10000;
+}
+
```

Comment:
Add API header @brief @param

Reply:
added #3096600 to PR (I added the API header in .h file)